File: GrowthWare_CS/GrowthWareFramework/ModelObjects/MRoleProfile.cs
using System;
using System.Collections;
using System.Collections.Generic;
using System.Data;
using System.Diagnostics;
using System.Web;
using GrowthWare.Framework.ModelObjects.Base.Interfaces;

using GrowthWare.Framework.ModelObjects.Base;

namespace GrowthWare.Framework.ModelObjects
{
	[Serializable(), CLSCompliant(true)]
	public class MRoleProfile : MProfile
	{

#region "Member Properties"
		    private string m_DESCRIPTION = string.Empty;
		    private int m_IsSystem = 0;
		    private int m_IsSystemOnly = 0;
			private int m_SE_SEQ_ID = 1;
#endregion


#region "Protected Methods"
		protected new void init(DataRow Datarow)
		{
			base.init(Datarow);
			base.setString(ref m_DESCRIPTION, ref Datarow, "DESCRIPTION");
			base.setInt(ref m_IsSystem, ref Datarow, "IS_SYSTEM");
			base.setInt(ref m_IsSystemOnly, ref Datarow, "IS_SYSTEM_ONLY");
			base.setInt(ref m_ID, ref Datarow, "ROLE_SEQ_ID");
			base.setString(ref m_Name, ref Datarow, "NAME");
		}
#endregion

		#region "Public Methods"
		/// <summary>
		/// Will return a message profile with the default vaules
		/// </summary>
		/// <remarks></remarks>

		public MRoleProfile()
		{
		}

		public MRoleProfile(DataRow dr)
		{
			init(dr);
		}
		#endregion

		#region "Public Properties"
		public int SecurityEntityID {
			get { return m_SE_SEQ_ID; }
			set { m_SE_SEQ_ID = value; }
		}
		public string Description {
			get { return m_DESCRIPTION; }
			set { m_DESCRIPTION = value.Trim(); }
		}

		public int IsSystem {
			get { return m_IsSystem; }
			set { m_IsSystem = value; }
		}

		public int IsSystemOnly {
			get { return m_IsSystemOnly; }
			set { m_IsSystemOnly = value; }
		}
		#endregion
	}
}

File: GrowthWare_VB/GrowthWareFrameworkModel/Profiles/MRoleProfile.vb
Imports GrowthWare.Framework.ModelObjects.Base.Interfaces
Imports GrowthWare.Framework.ModelObjects.Base

Namespace ModelObjects
	<Serializable(), CLSCompliant(True)> _
	Public Class MRoleProfile
		Inherits MProfile
		Implements IProfile

#Region "Member Properties"
		Private mDESCRIPTION As String = String.Empty
		Private mIsSystem As Integer = 0
		Private mIsSystemOnly As Integer = 0
		Private mErrorCode As Integer = -1
		Private mSE_SEQ_ID As Integer = 1
#End Region

#Region "Protected Methods"
		Protected Overloads Sub Init(ByVal dr As DataRow)
			On Error Resume Next
			MyBase.Init(dr)
			mDESCRIPTION = CStr(dr("DESCRIPTION")).Trim
			mIsSystem = CInt(dr("IS_SYSTEM"))
			mIsSystemOnly = CInt(dr("IS_SYSTEM_ONLY"))
			m_ID = CInt(dr("ROLE_SEQ_ID"))
			m_Name = CStr(dr("NAME")).Trim
		End Sub
#End Region

#Region "Public Methods"
		''' <summary>
		''' Will return a message profile with the default vaules
		''' </summary>
		''' <remarks></remarks>
		Public Sub New()

		End Sub

		Public Sub New(ByVal dr As DataRow)
			Init(dr)
		End Sub
#End Region

#Region "Public Properties"
		Public Property SecurityEntityID() As Integer
			Get
				Return mSE_SEQ_ID
			End Get
			Set(ByVal value As Integer)
				mSE_SEQ_ID = value
			End Set
		End Property
		Public Property Description() As String
			Get
				Return mDESCRIPTION
			End Get
			Set(ByVal value As String)
				mDESCRIPTION = value.Trim
			End Set
		End Property

		Public Property IsSystem() As Integer
			Get
				Return mIsSystem
			End Get
			Set(ByVal value As Integer)
				mIsSystem = value
			End Set
		End Property

		Public Property IsSystemOnly() As Integer
			Get
				Return mIsSystemOnly
			End Get
			Set(ByVal value As Integer)
				mIsSystemOnly = value
			End Set
		End Property
#End Region
	End Class
End Namespace